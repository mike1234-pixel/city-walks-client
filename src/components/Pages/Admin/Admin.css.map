{"version":3,"sourceRoot":"","sources":["Admin.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;;;AAGF;EACE;EACA;;;AAGF;EACE;;AACA;EACE;;;AAKF;EADF;IAEI","file":"Admin.css"}