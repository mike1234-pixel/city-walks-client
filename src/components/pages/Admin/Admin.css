.admin-portal {
  text-align: center;
  min-height: 50vh;
  background-image: url("../../../site-images//laptop-background.jpg");
  background-repeat: no-repeat;
  background-size: cover;
  background-attachment: fixed;
}

.admin-login-form {
  padding: 20px;
}

.admin-panel {
  background: var(--blue);
}
