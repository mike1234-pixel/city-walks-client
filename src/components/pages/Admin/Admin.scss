/** @format */

.admin-portal {
  text-align: center;
}

.admin-login-form {
  margin-top: 20px;
  margin-bottom: 20px;
}

.admin-panel {
  background: black;
  .nav-link {
    font-size: 12px;
  }
}

.navbar-collapse {
  @media (max-width: 767px) {
    text-align: left;
  }
}
