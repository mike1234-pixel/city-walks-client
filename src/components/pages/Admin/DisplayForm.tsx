import AddBlogPostForm from "./AddBlogPostForm/AddBlogPostForm";
import AddCityForm from "./AddCityForm/AddCityForm";
import AddWalkForm from "./AddWalkForm/AddWalkForm";
import DeleteBlogPostForm from "./DeleteBlogPostForm/DeleteBlogPostForm";
import DeleteBoardForm from "./DeleteBoardForm/DeleteBoardForm";
import DeleteCityForm from "./DeleteCityForm/DeleteCityForm";
import DeleteWalkForm from "./DeleteWalkForm/DeleteWalkForm";
import SetFeaturedWalkForm from "./SetFeaturedWalkForm/SetFeaturedWalkForm";

interface DisplayFormProps {
  form: string;
}

const DisplayForm = (props: DisplayFormProps) => {
  const { form } = props;

  switch (form) {
    case "addWalk":
      return <AddWalkForm />;
    case "addCity":
      return <AddCityForm />;
    case "setFeaturedWalk":
      return <SetFeaturedWalkForm />;
    case "deleteWalk":
      return <DeleteWalkForm />;
    case "deleteCity":
      return <DeleteCityForm />;
    case "addBoard":
      return <DeleteBoardForm />;
    case "deleteBoard":
      return <DeleteBoardForm />;
    case "addBlogPost":
      return <AddBlogPostForm />;
    case "deleteBlogPost":
      return <DeleteBlogPostForm />;
    default:
      return <p>error: could not find form</p>;
  }
};

export default DisplayForm;
