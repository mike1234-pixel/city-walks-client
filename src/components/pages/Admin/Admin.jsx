import { useState, useEffect } from "react"
import { MDBBtn, MDBNavbar, MDBNavbarBrand, MDBNavbarNav, MDBNavItem, MDBNavLink, MDBHamburgerToggler, MDBCollapse, MDBIcon, MDBInput } from "mdbreact";
import axios from "axios"
import qs from "qs"
import AddWalkForm from './AddWalkForm/AddWalkForm'
import AddCityForm from './AddCityForm/AddCityForm'
import SetFeaturedWalkForm from './SetFeaturedWalkForm/SetFeaturedWalkForm'
import DeleteWalkForm from './DeleteWalkForm/DeleteWalkForm'
import DeleteCityForm from './DeleteCityForm/DeleteCityForm'
import AddBoardForm from './AddBoardForm/AddBoardForm'
import DeleteBoardForm from "./DeleteBoardForm/DeleteBoardForm"
import AddBlogPostForm from "./AddBlogPostForm/AddBlogPostForm"
import DeleteBlogPostForm from "./DeleteBlogPostForm/DeleteBlogPostForm"
import './Admin.css'

const Admin = () => {

    const [form, setForm] = useState("")
    const [toggleAdminPanel, setToggleAdminPanel] = useState(false)

    const handleToggleAdminPanel = () => {
      setToggleAdminPanel(!toggleAdminPanel)
    }

    let displayForm
    if (form === "addWalk") {      
        displayForm = <AddWalkForm/>;    
    } else if (form === "addCity") {      
        displayForm = <AddCityForm/>;
    } else if (form === "setFeaturedWalk") {
        displayForm = <SetFeaturedWalkForm/>
    } else if (form === "deleteWalk") {
        displayForm = <DeleteWalkForm/>
    } else if (form === "deleteCity") {
        displayForm = <DeleteCityForm/>
    } else if (form === "addBoard") {
        displayForm = <AddBoardForm/>
    } else if (form === "deleteBoard") {
        displayForm = <DeleteBoardForm/>
    } else if (form === "addBlogPost") {
        displayForm = <AddBlogPostForm/>
    } else if (form === "deleteBlogPost") {
      displayForm = <DeleteBlogPostForm/>
    }

    useEffect(() => {
        window.scrollTo(0, 0);
      });
    
    const [adminUserName, setAdminUserName] = useState("")
    const [adminPassword, setAdminPassword] = useState("")
    const [adminLoggedIn, setAdminLoggedIn] = useState(false)

    const handleSubmitAdminLogin = (event) => {
        event.preventDefault()
  
        const payload = {
            username: adminUserName,
            password: adminPassword,
          };
  
    axios
        .post("http://localhost:5000/admin-login", qs.stringify(payload))
        .then((res, err) => {
          if (err) {
            console.log(err);
          } else if (res.data === "unsuccessful login attempt") {
            alert("Unsuccessful Login Attempt. Please Try Again.")
          } else {
            alert("Successfully Logged In")
            setAdminLoggedIn(true)
            setAdminUserName("")
            setAdminPassword("")
            window.scrollTo(0, 0)
          }
        });

    }

    const handleChangeAdminLogin = (event) => {
        switch(event.target.name) {
            case "admin-username":
              setAdminUserName(event.target.value)
              break;
            case "admin-password":
              setAdminPassword(event.target.value)
              break;
        }
    }

    const adminLogOut = () => {
        setAdminLoggedIn(false)
    }

    return (
        <div className="admin-portal">
            {adminLoggedIn ? 
            <div>
              <MDBNavbar className="admin-panel" dark expand="md">
                <MDBNavbarBrand>
                  <strong className="white-text">Admin Portal</strong>
                </MDBNavbarBrand>
                <MDBHamburgerToggler color="#fff" className="hamburger1" id="hamburger2" onClick={handleToggleAdminPanel} />
                  <MDBCollapse id="navbarCollapse3" isOpen={toggleAdminPanel} navbar>
                  <MDBNavbarNav left>
                    <MDBNavItem>
                      <MDBNavLink to="#!" onClick={() => setForm("addWalk")}>Add Walk</MDBNavLink>
                    </MDBNavItem>
                    <MDBNavItem>
                      <MDBNavLink to="#!" onClick={() => setForm("addCity")}>Add City</MDBNavLink>
                    </MDBNavItem>
                    <MDBNavItem>
                      <MDBNavLink to="#!" onClick={() => setForm("setFeaturedWalk")}>Set Featured Walk</MDBNavLink>
                    </MDBNavItem>
                    <MDBNavItem>
                      <MDBNavLink to="#!" onClick={() => setForm("deleteWalk")}>Delete Walk</MDBNavLink>
                    </MDBNavItem>
                    <MDBNavItem>
                      <MDBNavLink to="#!" onClick={() => setForm("deleteCity")}>Delete City</MDBNavLink>
                    </MDBNavItem>
                    <MDBNavItem>
                      <MDBNavLink to="#!" onClick={() => setForm("addBoard")}>Add Board</MDBNavLink>
                    </MDBNavItem>
                    <MDBNavItem>
                      <MDBNavLink to="#!" onClick={() => setForm("deleteBoard")}>Delete Board</MDBNavLink>
                    </MDBNavItem>
                    <MDBNavItem>
                      <MDBNavLink to="#!" onClick={() => setForm("addBlogPost")}>Add Blog Post</MDBNavLink>
                    </MDBNavItem>
                    <MDBNavItem>
                      <MDBNavLink to="#!" onClick={() => setForm("deleteBlogPost")}>Delete Blog Post</MDBNavLink>
                    </MDBNavItem>
                    </MDBNavbarNav>
                    <MDBNavbarNav right>
                    <MDBNavItem>
                      <MDBNavLink to="#!" onClick={() => adminLogOut()}>Admin Logout</MDBNavLink>
                    </MDBNavItem>
                    </MDBNavbarNav>
                  </MDBCollapse>
              </MDBNavbar>
            <br/>
            <br/>
            <br/>
            {displayForm} 
            </div> : 
            <form className="admin-login-form" onSubmit={handleSubmitAdminLogin}>
            <MDBInput type="text" name="admin-username" id="admin-username" value={adminUserName} label="admin username" onChange={handleChangeAdminLogin} maxLength="70" required/>
            <MDBInput type="password" name="admin-password" id="admin-password" value={adminPassword} label="admin password" onChange={handleChangeAdminLogin} maxLength="70" required/>
            <MDBBtn outline color="elegant" type="submit">Administrator Login <MDBIcon icon="sign-in-alt" /></MDBBtn>
            </form>
            }
        </div>
    )
}

export default Admin