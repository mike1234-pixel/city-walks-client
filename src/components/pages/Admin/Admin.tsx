import React, { ReactNode, useState, useEffect } from "react"
import { MDBBtn, MDBNavbar, MDBNavbarBrand, MDBNavbarNav, MDBNavItem, MDBNavLink, MDBHamburgerToggler, MDBCollapse, MDBIcon, MDBInput, MDBContainer } from "mdbreact";
import axios, { AxiosResponse } from "axios"
import qs from "qs"
import AddWalkForm from './AddWalkForm/AddWalkForm'
import AddCityForm from './AddCityForm/AddCityForm'
import SetFeaturedWalkForm from './SetFeaturedWalkForm/SetFeaturedWalkForm'
import DeleteWalkForm from './DeleteWalkForm/DeleteWalkForm'
import DeleteCityForm from './DeleteCityForm/DeleteCityForm'
import AddBoardForm from './AddBoardForm/AddBoardForm'
import DeleteBoardForm from "./DeleteBoardForm/DeleteBoardForm"
import AddBlogPostForm from "./AddBlogPostForm/AddBlogPostForm"
import DeleteBlogPostForm from "./DeleteBlogPostForm/DeleteBlogPostForm"
import AdminCredentials from "../../../types/PostRequests/AdminCredentials";
import './Admin.scss'

const Admin: React.FC = () => {

  const [form, setForm] = useState<string>("")
  const [toggleAdminPanel, setToggleAdminPanel] = useState<boolean>(false)

  const handleToggleAdminPanel: () => void = () => {
    setToggleAdminPanel(!toggleAdminPanel)
  }

  let displayForm: ReactNode
  if (form === "addWalk") {
    displayForm = <AddWalkForm />;
  } else if (form === "addCity") {
    displayForm = <AddCityForm />;
  } else if (form === "setFeaturedWalk") {
    displayForm = <SetFeaturedWalkForm />
  } else if (form === "deleteWalk") {
    displayForm = <DeleteWalkForm />
  } else if (form === "deleteCity") {
    displayForm = <DeleteCityForm />
  } else if (form === "addBoard") {
    displayForm = <AddBoardForm />
  } else if (form === "deleteBoard") {
    displayForm = <DeleteBoardForm />
  } else if (form === "addBlogPost") {
    displayForm = <AddBlogPostForm />
  } else if (form === "deleteBlogPost") {
    displayForm = <DeleteBlogPostForm />
  }

  useEffect(() => {
    window.scrollTo(0, 0);
  });

  const [adminUserName, setAdminUserName] = useState<string>("")
  const [adminPassword, setAdminPassword] = useState<string>("")
  const [adminLoggedIn, setAdminLoggedIn] = useState<boolean>(false)

  const handleSubmitAdminLogin: (event: React.FormEvent) => void = (event) => {
    event.preventDefault()

    const payload: AdminCredentials = {
      username: adminUserName,
      password: adminPassword,
    };

    axios
      .post("https://city-walks.herokuapp.com/admin-login", qs.stringify(payload))
      .then((res: AxiosResponse) => {
        if (res.data === "unsuccessful login attempt") {
          alert("Unsuccessful Login Attempt. Please Try Again.")
        } else {
          alert("Successfully Logged In")
          setAdminLoggedIn(true)
          setAdminUserName("")
          setAdminPassword("")
          window.scrollTo(0, 0)
        }
      });

  }

  const handleChangeAdminLogin: (event: React.ChangeEvent<any>) => void = (event) => {
    switch (event.target.name) {
      case "admin-username":
        setAdminUserName(event.target.value)
        break;
      case "admin-password":
        setAdminPassword(event.target.value)
        break;
    }
  }

  const adminLogOut: () => void = () => {
    setAdminLoggedIn(false)
  }

  return (
    <div className="admin-portal min-page-height">
      {adminLoggedIn ?
        <div>
          <MDBNavbar className="admin-panel" dark expand="md">
            <MDBContainer>
              <MDBNavbarBrand>
                <strong className="white-text">Admin Portal</strong>
              </MDBNavbarBrand>
              <MDBHamburgerToggler color="#fff" className="hamburger1" id="hamburger2" onClick={handleToggleAdminPanel} />
              <MDBCollapse id="navbarCollapse3" isOpen={toggleAdminPanel} navbar>
                <MDBNavbarNav left>
                  <MDBNavItem>
                    <MDBNavLink to="#!" onClick={() => setForm("addWalk")}>Add Walk</MDBNavLink>
                  </MDBNavItem>
                  <MDBNavItem>
                    <MDBNavLink to="#!" onClick={() => setForm("addCity")}>Add City</MDBNavLink>
                  </MDBNavItem>
                  <MDBNavItem>
                    <MDBNavLink to="#!" onClick={() => setForm("setFeaturedWalk")}>Set Featured Walk</MDBNavLink>
                  </MDBNavItem>
                  <MDBNavItem>
                    <MDBNavLink to="#!" onClick={() => setForm("deleteWalk")}>Delete Walk</MDBNavLink>
                  </MDBNavItem>
                  <MDBNavItem>
                    <MDBNavLink to="#!" onClick={() => setForm("deleteCity")}>Delete City</MDBNavLink>
                  </MDBNavItem>
                  <MDBNavItem>
                    <MDBNavLink to="#!" onClick={() => setForm("addBoard")}>Add Board</MDBNavLink>
                  </MDBNavItem>
                  <MDBNavItem>
                    <MDBNavLink to="#!" onClick={() => setForm("deleteBoard")}>Delete Board</MDBNavLink>
                  </MDBNavItem>
                  <MDBNavItem>
                    <MDBNavLink to="#!" onClick={() => setForm("addBlogPost")}>Add Blog Post</MDBNavLink>
                  </MDBNavItem>
                  <MDBNavItem>
                    <MDBNavLink to="#!" onClick={() => setForm("deleteBlogPost")}>Delete Blog Post</MDBNavLink>
                  </MDBNavItem>
                </MDBNavbarNav>
                <MDBNavbarNav right>
                  <MDBNavItem>
                    <MDBNavLink to="#!" onClick={() => adminLogOut()}>Admin Logout</MDBNavLink>
                  </MDBNavItem>
                </MDBNavbarNav>
              </MDBCollapse>
            </MDBContainer>
          </MDBNavbar>
          <br />
          <br />
          <br />
          <MDBContainer>
            {displayForm}
          </MDBContainer>
        </div> :
        <MDBContainer>
          <form className="admin-login-form" onSubmit={handleSubmitAdminLogin}>
            <MDBInput type="text" name="admin-username" id="admin-username" value={adminUserName} label="admin username" onChange={handleChangeAdminLogin} maxLength="70" required />
            <MDBInput type="password" name="admin-password" id="admin-password" value={adminPassword} label="admin password" onChange={handleChangeAdminLogin} maxLength="70" required />
            <MDBBtn outline color="elegant" type="submit">Administrator Login <MDBIcon icon="sign-in-alt" /></MDBBtn>
          </form>
        </MDBContainer>
      }
    </div>

  )
}

export default Admin