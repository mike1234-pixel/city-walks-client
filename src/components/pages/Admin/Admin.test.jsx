import renderer from 'react-test-renderer'
import Admin from "./Admin"

// snapshot test
it('Admin Page renders correctly', () => {  

  const tree = renderer
    .create(
        <Admin />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});