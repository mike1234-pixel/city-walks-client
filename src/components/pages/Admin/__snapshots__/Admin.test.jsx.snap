// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Page renders correctly 1`] = `
<div
  className="admin-portal"
>
  <form
    className="admin-login-form"
    onSubmit={[Function]}
  >
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="admin-username"
        maxLength="70"
        name="admin-username"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="admin-username"
        id=""
        onClick={[Function]}
      >
        admin username
      </label>
    </div>
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="admin-password"
        maxLength="70"
        name="admin-password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="password"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="admin-password"
        id=""
        onClick={[Function]}
      >
        admin password
      </label>
    </div>
    <button
      className="btn-elegant btn-outline-elegant btn Ripple-parent"
      data-test="button"
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="submit"
    >
      Administrator Login 
      <i
        className="fa fa-sign-in-alt"
        data-test="fa"
      />
      <div
        className="Ripple Ripple-outline "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </button>
  </form>
</div>
`;
